lms/djangoapps/course_api/serializers.py

```diff
@@ -159,6 +160,16 @@ def get_overview(self, course_overview):
         # from the CourseOverview object in SQL.
         return CourseDetails.fetch_about_attribute(course_overview.id, 'overview')
 
+    def to_representation(self, instance):
+        """
+        Get the `certificate_available_date` in response
+        if `auto_certificate_generation` is enabled
```

Comment:
This will also help in refactoring in future if the switch name changes.

```suggestion
        if the `certificates.auto_certificate_generation` switch is enabled
```